lambdas/raster_tiler/lambda_function.py

```diff
@@ -37,15 +37,42 @@ def array_to_img(arr: np.ndarray) -> str:
 
 def get_tile_array(src_tile: str, window: Window) -> np.ndarray:
     """Create mercator tile from GFW WM Tile Set images."""
-
-    with rasterio.open(src_tile) as src:
-        profile = src.profile
-        bands = profile["count"]
-        indexes = tuple(range(1, bands + 1))
-        out_shape = (len(indexes), TILE_SIZE, TILE_SIZE)
-        data = src.read(
-            window=window, boundless=True, out_shape=out_shape, indexes=indexes
-        )
+    # if running lambda in localstack, need to use special docker IP address provided in env to reach localstack
+    LOCALSTACK_HOSTNAME = os.environ.get("LOCALSTACK_HOSTNAME", None)
+    AWS_ENDPOINT_HOST = f"{LOCALSTACK_HOSTNAME}:4566" if LOCALSTACK_HOSTNAME else None
+
+    # FIXME this is just temporary to make test work, some issue with reading ranges in localstack S3
+    #
+    # import boto3
+    # s3_client = boto3.client("s3", endpoint_url=f"http://{AWS_ENDPOINT_HOST}")
+    #
+    # from urllib.parse import urlparse
+    # def get_s3_path_parts(path):
+    #     parsed = urlparse(path)
+    #     bucket = parsed.netloc
+    #     key = parsed.path.lstrip("/")
+    #     return bucket, key
+    #
+    # with open('/tmp/raster.tif', 'wb') as file:
+    #     bucket, key = get_s3_path_parts(src_tile)
+    #     print(bucket)
+    #     print(key)
+    #     s3_client.download_fileobj(bucket, key, file)
+
+    with rasterio.Env(
+            AWS_HTTPS='NO',
+            AWS_VIRTUAL_HOSTING=False,
+            AWS_S3_ENDPOINT=AWS_ENDPOINT_HOST,
+            GDAL_DISABLE_READDIR_ON_OPEN='YES'
+    ):
+        with rasterio.open(src_tile) as src:
+            profile = src.profile
+            bands = profile["count"]
+            indexes = tuple(range(1, bands + 1))
+            out_shape = (len(indexes), TILE_SIZE, TILE_SIZE)
+            data = src.read(
+                window=window, boundless=True, out_shape=out_shape, indexes=indexes
```

Comment:
instead of nesting two `with` statements you can combine them and seperate with a comma
```suggestion

    GDAL_ENV = {
        "AWS_HTTPS": "NO",
        "AWS_VIRTUAL_HOSTING": False,
        "AWS_S3_ENDPOINT": AWS_ENDPOINT_HOST,
        "GDAL_DISABLE_READDIR_ON_OPEN": "YES"
        }
            
    with rasterio.Env(**GDAL_ENV), rasterio.open(src_tile) as src:
        profile = src.profile
        bands = profile["count"]
        indexes = tuple(range(1, bands + 1))
        out_shape = (len(indexes), TILE_SIZE, TILE_SIZE)
        data = src.read(
            window=window, boundless=True, out_shape=out_shape, indexes=indexes
```